public_html/wp-content/plugins/pattern-creator/src/components/welcome-guide/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+import { Guide } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as editPostStore } from '@wordpress/edit-post';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ImageCollectionImage, PatternEditorImage, PatternsImage } from './images';
+import './style.css';
+
+/**
+ * Module constants
+ */
+const GUIDE_ID = 'welcomeGuide';
```

Comment:
Does this conflict with the core welcome guide? maybe it should be `wporgPatternGuide`?

Reply:
It doesn't but it's a good idea.

Reply:
Actually, it is the same as core to replace it completely in the editor. This makes sure it loads on page load and when users click `Welcome Guide` in the menu. We could explicitly remove core's version and add ours but this seems like _enough_. Thoughts?

Reply:
After further investigation, now that the editor is initialized differently, we'll have to turn off the welcome guide if possible. There's a small flash of core's version.

Reply:
It turns out you can't really remove the welcome guide (you can stop it from showing for new users on load, but the welcome guide menu link will still exist and call the core guide) but you can replace it by registering a `welcomeGuide` on top of it. 

Now, since we now use `registerPlugin`, which appears to have less priority than we used to have with our custom-built editor, there is a very slight shift from the old one to the new one (see below). The shift is pretty slight but exists nonetheless. The obvious downside of this approach is that if core changes the `id` of the welcome guide, this will no longer work... although this will fail pretty silently.


![](https://d.pr/i/VWNFcu.gif)